context/config_file.go

```diff
@@ -67,14 +68,91 @@ func isLegacy(root *yaml.Node) bool {
 	return true
 }
 
+func migrateConfig(fn string, root *yaml.Node) error {
+	// TODO i'm sorry
+	newConfigData := map[string]map[string]map[string]string{}
+	newConfigData["hosts"] = map[string]map[string]string{}
```

Comment:
protip: you could alias some of these types so it looks more expressive. example:
```go
type ConfigEntry map[string]string
type ConfigMap map[string]ConfigEntry

newConfigData := make(ConfigMap)
```
```